LochNessBuilder/BuilderRegistry.cs

```diff
@@ -62,7 +62,7 @@ private static void RegisterMostAppropriateGetter(Type builderType)
             if (!propsReturningABuilder.Any())
             {
                 throw new NotImplementedException(
```

Comment:
We should probably consider catch these exceptions on a per-type basis in the static constructor - as if we don't, any dodgy code makes this class throw exceptions on use https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/static-constructors

If we *wish* to throw an exception on use - because if someone's using WithBuilt then they are happy with this - then we should document that behaviour.

My gut is that we should instead throw an exception when you try to use `WithBuilt` for this specific type?

Reply:
Ah. I see what you mean. That surprises me; I expected the exception itself to get thrown directly out, at the point that the static ctor gets triggered. Ah well.

So we have 3 options:
 * Leave it as is, and any attempt to *Build* a builder that has `.WithBuilt()` will blow up.
 * Create a Token "BuilderRegistry.ValidateFactories()" static method, which we invoke explicitly in `.WithBuilt()` to force the exception to happen at configuration time.
 * Remove this exception entirely (i.e. just ignore the fact that someone has mis-used the `[BuilderFactory]` attribute)

Note that we can't "throw an exception when you try to use WithBuilt for this specific type?" because there is no specific type - we don't know what kind of `Builder` the User intended this `BuilderFactory` to provide.

Reply:
Oh I see. I think we should remove the exception entirely.

I also think we should add some try/catches around various parts of the static constructor's call stack to prevent the whole thing blowing up if there are any exceptions we didn't foresee.

(EG around RegisterMostAppropriateGetter, and also around the whole static constructor).

Reply:
I'm happy removing the "no Builders found" exception. But I'm less keen on suppressing all exceptions entirely. If there is some edge case that can throw errors, then suppressing them will cause this to all silently and undiagnosably fail.

We could catch it and store the error, and then re-throw it in .Resolve() ... but I'm not convinced that the existing behaviour is *actually* different from that in any way taht's meaningful?